nx.json

```diff
@@ -45,46 +45,63 @@
   "targetDefaults": {
     "build": {
       "dependsOn": ["^build"],
-      "inputs": ["production", "^production"]
+      "inputs": ["buildInputs", "^buildInputs"]
     },
     "test": {
-      "inputs": ["default", "^production", "{workspaceRoot}/karma.conf.js"]
+      "inputs": ["testInputs", "^buildInputs"]
     },
     "e2e": {
-      "inputs": ["default", "^production"]
+      "inputs": ["e2eInputs", "^buildInputs"]
     },
     "lint": {
-      "inputs": ["default", "{workspaceRoot}/.eslintrc.json"]
+      "inputs": ["lintInputs"]
     },
     "build-storybook": {
-      "inputs": [
-        "default",
-        "^production",
-        "{workspaceRoot}/.storybook/**/*",
-        "!{projectRoot}/.storybook/**/*"
-      ]
+      "inputs": ["storybookInputs", "^buildInputs"]
     }
   },
   "namedInputs": {
-    "default": ["{projectRoot}/**/*", "sharedGlobals"],
-    "sharedGlobals": [
+    "buildInputs": [
```

Comment:
Do all of these inputs include changes to the `package.json`, `package-lock.json`, and `tsconfig.base.json` files, as well as the `/scripts` directory? All four of these areas can have an impact on `build`, `test`, `lint`, and `e2e`, so we should probably run the full suite if any of them change.

Reply:
NX handles `package.json` and `tsconfig.base.json`. I'll add `scripts`.